pwem/protocols/protocol_set_editor.py

```diff
@@ -51,31 +51,53 @@ def _defineParams(self, form):
                       pointerClass='EMSet',
                       label='Set to edit',
                       help='Set which items will be modified.')
-        # formula
+        # formula for main items
         form.addParam('formula', params.StringParam, label="Formula",
-                      help='A python code compatible with eval, where item represents each of '
+                      help='Any python code compatible with eval, where item represents each of '
                            'the elements of the set. E.g.: item._resolution.set(item._resolution.get() +1).'
                            'You could also use modules like "import numpy;  item._resolution .... "')
 
+        # formula for subitems items
+        form.addParam('subitemsformula', params.StringParam, label="Sub item formula",
+                      help='Only valid for complex sets with subelements: Classes -> Particles, TiltSeries -> TiltImage, ..'
+                           'Any python code compatible with eval, where item represents each of '
+                           'the SUBELEMENTS each ITEM. E.g.: subitem.tiltAngle.set(subitem.tiltAngle.get() +1).'
+                           'You could also use modules like "import numpy;  subitem._resolution .... "')
+
     def _insertAllSteps(self):
-        self._insertFunctionStep('formulaStep')
+        self._insertFunctionStep(self.formulaStep)
 
     def formulaStep(self):
         """
         Goes through all items in the input set and applies the formula to each of them using exec.
         Complex python code could be run separating lines with ;  To use numpy you could do
         import numpy; item._resolution.set(numpy.random.randint(10))
         """
+
+        import time
+        time.sleep(10)
```

Comment:
Why is this needed?

Reply:
To see if you review the code!! Was intentional!! 😜